### PR TITLE
Show only not empty events and jobs

### LINKED ISSUE #175: SHIFT SIGN UP list enchancement: show only not empty events and jobs
In the "SHIFT SIGN UP" list we can see all jobs and events. Even events and jobs without shifts. Show only not empty events and jobs.

image

### PR DESCRIPTION
**Pull request by ahmedsabie**
closes #175 
Volunteers will only be able to see events with jobs (at least 1 job must have a shift), and jobs with shifts.

What admin sees:
admin

What volunteer sees:
user

### CODE REVIEW

File: vms/event/services.py [hunk 1]
```diff
@@ -89,3 +89,15 @@ def get_events_by_date(start_date, end_date):
 def get_events_ordered_by_name():
     event_list = Event.objects.all().order_by('name')
     return event_list
+
+
+def remove_empty_events(event_list):
+    """ Removes all events from an event list without jobs or shifts """
+
```
Comment (tapasweni-pathak) [inline vms/event/services.py]: There should be no blank line either before or after the docstring. Can you please remove this @ahmedsabie?

**Comment by ahmedsabie** [comment]
@tapasweni-pathak is there a way to push that change without adding a new commit? or should I just put in "Remove blank lines around docstrings"

**Comment by ahmedsabie** [comment]
@tapasweni-pathak ok I soft reset, pushed the changes, and reopened the pull request to add only 1 new commit

**Comment by ahmedsabie** [comment]
@vubo great catch :+1: ! I forgot to even check what shift_list was doing in the first place to the shifts. I added the necessary changes, and improved the new function names :)

**Comment by vubo** [comment]
@ahmedsabie now it looks good! I added a new unit-tests-issue for your new functions: https://github.com/systers/vms/issues/187
@tapasweni-pathak can we add this issue also to the GCI? There is an easy one.

**Comment by tapasweni-pathak** [comment]
@ahmedsabie Thanks for your work. :) 
@vubo Done. :+1: :smile:

**Comment by vubo** [comment]
The catch one more time (my deleted comment):
I tested it and it doesn't work, because you use get_shifts_by_job_id, but for list_shifts_sign_up in /shift/list_shifts_sign_up/ the function get_shifts_with_open_slots_for_volunteer https://github.com/systers/vms/blob/develop/vms/shift/views.py#L482 is used. So you can have some shifts in get_shifts_by_job_id, but no shifts in get_shifts_with_open_slots_for_volunteer.

@ahmedsabie to test your code you should write a unit test for every new function, that you implemented. It can help you to debug your code.

Unit tests for jobs: https://github.com/systers/vms/blob/develop/vms/job/tests.py
Unit tests for events: https://github.com/systers/vms/blob/develop/vms/event/tests.py

I recommend you to program unit tests to avoid most of the bugs and submit this tests with your new functions in the PR.

**Comment by ahmedsabie** [comment]
@vubo will do that! I can see the format from the other unit tests. Are there any general tips or things I should watch out for when making tests?

**Comment by vubo** [comment]
@ahmedsabie  for unit testing we are using the unittest framework for Python: https://docs.python.org/3/library/unittest.html - you can read this and also you can see examples in our unit tests. Good luck! :four_leaf_clover:

**Comment by ahmedsabie** [comment]
@vubo The other unit tests were pretty helpful! Certainly made making the test cases a lot easier and logical ~ pull request: #188  :)

**Pull request merged**